tests/pytests/integration/states/test_file.py

```diff
@@ -346,3 +415,594 @@ def test_issue_60203(
+    # Create a new unpatched set of files
+    os.makedirs(patch_file_dest / "foo" / "bar")
+    numbers_file = patch_file_dest / "foo" / "numbers.txt"
+
+    sls_patch_contents = """
+        do-patch:
+          file.patch:
+            - name: {numbers_file}
+            - source: {numbers_patch}
+        """.format(
+        numbers_file=numbers_file, numbers_patch=numbers_patch_file
+    )
+
+    sls_patch_tempfile = temp_file(
+        "test_patch.sls", sls_patch_contents, patch_file_dest
+    )
+    numbers_tempfile = temp_file(numbers_file, content[0], patch_file_dest)
+
+    with sls_patch_tempfile, numbers_tempfile:
+        # Test application with test=True mode
+        ret = salt_call_cli.run("state.apply", "test_patch", test=True)
+        assert ret.exitcode == 0
+        assert ret.json
+        state_run = next(iter(ret.json.values()))
+        assert state_run["result"] is None
+        assert state_run["comment"] == "The patch would be applied"
+
+        # Apply the patch for real. We'll then be able to test below that we
+        # exit with a True rather than a None result if test=True is used on an
+        # already-applied patch.
+        ret = salt_call_cli.run("state.apply", "test_patch")
+        assert ret.exitcode == 0
+        assert ret.json
+
+        # Check to make sure the patch was applied okay
+        state_run = next(iter(ret.json.values()))
+        assert state_run["result"] is True
+        assert state_run["comment"] == "Patch successfully applied"
+
+        # Run again with test=True. Since the pre-check happens before we do
+        # the __opts__['test'] check, we should exit with a True result just
+        # the same as if we try to run this state on an already-patched file
+        # *without* test=True.
+        ret = salt_call_cli.run("state.apply", "test_patch", test=True)
+        assert ret.exitcode == 0
+        assert ret.json
+
+        # Check to make sure the patch was applied okay
+        state_run = next(iter(ret.json.values()))
+        assert state_run["result"] is True
+        assert state_run["comment"] == "Patch was already applied"
+
+        # Empty the file to ensure that the patch doesn't apply cleanly
+        with salt.utils.files.fopen(numbers_file, "w"):
+            pass
+
+        # Run again with test=True. Similar to the above run, we are testing
+        # that we return before we reach the __opts__['test'] check. In this
+        # case we should return a False result because we should already know
+        # by this point that the patch will not apply cleanly.
+        ret = salt_call_cli.run("state.apply", "test_patch", test=True)
+        assert ret.exitcode == 1
+        assert ret.json
+
+        # Check to make sure the patch was applied okay
+        state_run = next(iter(ret.json.values()))
+        assert state_run["result"] is False
+        assert "Patch would not apply cleanly" in state_run["comment"]
```

Comment:
These newly added tests, they don't really need a minion running right? I'm this case these could have been functional tests.
Also, why restrict them to only run on windows?

Reply:
It seems okay to me to have them as integration because they are calling `salt-call`.

Reply:
Yes, they are calling `salt-call` but that's not the intention of the tests, is it? To test `salt-call`... That's why I added my comment, we're adding points of test failure which are outside the scope of the test. Previously, we had to, these days we don't.

Reply:
With that said, I'm not saying the PR should be blocked, but we should turn them into functional, as soon as possible.